lib/types/string/index.js

```diff
@@ -351,45 +354,32 @@ internals.String = class extends Any {
                 let version = guidOptions.version[i];
                 Hoek.assert(typeof version === 'string', 'version at position ' + i + ' must be a string');
                 version = version.toLowerCase();
-                Hoek.assert(uuids[version], 'version at position ' + i + ' must be one of ' + Object.keys(uuids).join(', '));
-                Hoek.assert(versions.indexOf(version) === -1, 'version at position ' + i + ' must not be a duplicate.');
-                versions.push(version);
+                const versionNumber = uuids[version];
+                Hoek.assert(versionNumber, 'version at position ' + i + ' must be one of ' + Object.keys(uuids).join(', '));
+                Hoek.assert(!(versionNumber in versions), 'version at position ' + i + ' must not be a duplicate.');
```

Comment:
`hasOwnProperty` is usually faster, but that'd be marginal. Think a `Set` would do better ?